plutarch-test/src/Plutarch/Test.hs

```diff
@@ -0,0 +1,205 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE ImpredicativeTypes #-}
```

Comment:
I thought you wanted to use GHC 8.10 with this.

Reply:
It is needed for the `a` in `goldens :: PlutarchGolden -> [(String, ClosedTerm a)] `

Reply:
That is true and is quite unfortunate. You can have a newtype wrapper for `ClosedTerm` if you want, i.e.
```haskell
newtype MyClosedTerm a = MyClosedTerm (ClosedTerm a)
```
which should fix it AFAICT, and then you could do `MyClosedTerm $ ...` in the list you pass to `goldens`. You could also hard-code it to be `POpaque`. It's quite unfortunate that we still support GHC 8.10, but we will probably drop it once GHC 9.2.2 comes out, since I've realised that there isn't much point to the older GHC 8.10 support.